backends-common/postgres/src/test/java/org/apache/james/backends/postgres/SinglePostgresConnectionFactory.java

```diff
@@ -0,0 +1,42 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.postgres;
+
+import java.util.Optional;
+
+import org.apache.james.backends.postgres.utils.JamesPostgresConnectionFactory;
+import org.apache.james.core.Domain;
+
+import io.r2dbc.spi.Connection;
+import reactor.core.publisher.Mono;
+
+public class SinglePostgresConnectionFactory implements JamesPostgresConnectionFactory {
```

Comment:
Why?

Reply:
It is for test package
In imap mpt postgres module, the tests create a lot of new connection, that make throw exception. (too many connection from postgresql)

Reply:
How many is too moch connection?

Are you leaking - not closing the connection where you should?

And finally would it make sense to *share* and *reuse* the connection?

I would prefer this a **lot** rather than hacking your own connection factory for test, distinct from the one used in prod.

Reply:
My mistake when creating a new `JamesPostgresConnectionFactory` for each test.

Fixed